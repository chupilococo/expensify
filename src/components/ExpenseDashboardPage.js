import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseTotal from './ExpensesTotal';

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseTotal />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
