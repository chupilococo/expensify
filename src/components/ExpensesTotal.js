import numeral from 'numeral';
import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';


export const ExpenseTotal = (props) => {
  const cantidad = props.expenses.length;

  const total = props.expenses
    .map((expense) => expense.amount)
    .reduce((sum, val) => (sum + val), 0);

  return (
    <div>
      {
        cantidad === 0 ? null : (
          <h1>Viewing {cantidad} {cantidad === 1 ? 'expnese' : 'expneses'} totaling {numeral(total / 100).format('$0,0.00')}</h1>
        )
      }
    </div>
  );

}

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseTotal);
