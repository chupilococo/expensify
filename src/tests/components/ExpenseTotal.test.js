import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseTotal } from '../../components/ExpensesTotal'
import expenses from '../fixtures/expenses'


test('should render expenses totoal with multiple expenses', () => {
    const wrapper = shallow(<ExpenseTotal expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
})

test('should render expenses list with one expenses', () => {
    const wrapper = shallow(<ExpenseTotal expenses={[expenses[0]]} />);
    expect(wrapper).toMatchSnapshot();
})

test('should render expenses list with no expenses', () => {
    const wrapper = shallow(<ExpenseTotal expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
})