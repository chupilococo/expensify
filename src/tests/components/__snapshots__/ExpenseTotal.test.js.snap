// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses list with no expenses 1`] = `<div />`;

exports[`should render expenses list with one expenses 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expnese
     totaling 
    $1.95
  </h1>
</div>
`;

exports[`should render expenses totoal with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expneses
     totaling 
    $1,141.95
  </h1>
</div>
`;
